### ISSUE TITLE
Compatibility with OpenSearch 2.4

### ISSUE THREAD

**Issue by pmarki** [klevu/productsearch#76]
Hi,
We run latest klevu on Magento 2.4.6 and opensearch 2.4, search result page doesn't return any results but it works once klevu modules are removed. Could maybe someone confirm compatibility with Magento 2.4.6 and OpenSearch 2.4?
Many thanks,
Piotr

**Comment by sandipklevu** [comment]
Hi Piotr,

Thank you for raising this concern with us.
Our development team is already working on compatibility with Magento 2.4.6 and OpenSearch 2.4.
We will soon release the latest version.

Reach out to us [support@klevu.com](mailto:support@klevu.com) we will assist further.

**Comment by pmarki** [comment]
For anyone that will have the same issue before Klevu will provide a fix:
```
--- ./vendor/klevu/module-productsearch/Plugin/Framework/Search/Request/CleanerPlugin.php	
+++ ./vendor/klevu/module-productsearch/Plugin/Framework/Search/Request/CleanerPlugin.php	
@@ -332,6 +332,9 @@
             case "solr":
                 return '_id';
                 break;
+            case "opensearch":
+                return '_id';
+                break;
             default:
                 return 'entity_id';
                 break;
```

**Comment by klevu** [comment]
@pmarki thank you for opening this issue.
Today we have released 2.10.2 and 3.2.2 which includes this fix.

**Comment by klevu** [comment]
@pmarki Compatibility with OpenSearch has been added to the latest releases 2.11.0 and 3.3.0.
Thank you again for flagging this.
